### PR TITLE
feedback to "user interface"

### PR DESCRIPTION
**Pull request by thsvr**
Hi,

Can you please check our code and give feedback?

Thank you in advance :)

Thaís and Sofi

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env ruby
+# Welcome to players
+    def welcome
+    puts "Welcome Players!"
+    puts "Tic Tac Toe is a game for 2 players."
+    puts "The game is played on a grid that's 3 x 3"
+    puts "If you are X, your friend? is O"
+    puts "Players take turns putting their marks in empty squares"
```
Comment (samgaco) [inline bin/main.rb:13]: this should be also mocked in here

* where is the game asking for a move to the players? 
* What are the possible moves available?
* if a player inputs a move that is not valid, how is the game going to react?

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @SofiBretz and @thsvr :high_brightness: 

- [ ] We should be able to execute the game `ruby bin/main.rb` and see how it would look like in the console. (Of course, no real logic behind the things prompted in there). 

_Example: we execute, and we see the game asking us questions like: what is your name?, the game board being prompted and questions like, where do you want to place your move?, etc.  All the game from start to end should be mocked._

- [ ] I left some comments on things that should be mocked as well.


------------------


* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

**Comment by SofiBretz** [comment]
### Summary of changes made

- Linters fixed, no conflicts on this branch.

- New classes added to organized better the code of the game.

- The code is cleaner to read.

- Code executable.

**Review by Ispirett — CHANGES_REQUESTED** [review]
Hey!! @SofiBretz how are you?  great work @SofiBretz  @thsvr  setting up the folder structure  and welcome message  :+1:  :100: 

For this milestone, you need to have mock logic for the game. Thses are the things you should have below.

**[OPTIONAL]** 
- [ ] game ask players for their names

 **[REQUIRED]**
- [ ] game tells which player turn it is

- [ ]  game asks player to select from available moves

 - [ ]  game informs the player if the selected move is invalid

- [ ]   game displays board after the player move

- [ ] game informs the player if the selected move is a winning move

- [ ] game informs player if selected move is a draw move

- [ ]  game repeats all actions for next player's move

You don't  have to implement actual logic  you can use a ` puts "message" `
This step is to help you think through the structure of your code for the game.
you can use  a ` while loop ` to switch the turn of the players.

status: changes to be made

**Comment by SofiBretz** [comment]
### Summary of changes made

Linters fixed, no conflicts on this branch.

Advices of the tse corrected.

The code is cleaner to read.

In this file, the game is not working 100% because it is necessary to link the main.rb file with the other files that we have to complete on the milestone 3.

The point here is that in the main time when we were waiting for this review, we advanced with the game and now we finished, so please contact us if you want to see the entire code working because for this milestone we can not run it 100%. But we already finished the milestone 3.

**So please before request another thing contact us first.**  Thanks for understanding us, we made to much progress already.

File: bin/main.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+      puts 'Wow! It should be x or o, stay focus *_*'
+      element = gets.chomp.upcase!
+    end
+  end
+
+  def info_players
+    puts "\nOk! Player 1 give us ur name: "
+    name = gets.chomp
+    player1 = Player.new(name_valid?(name))
```
Comment (samgaco) [inline bin/main.rb:51]: the class is not defined

**Review by samgaco — APPROVED** [review]
Hi @SofiBretz and @thsvr 

The point of this milestone is to show the interface as it would show if the game were to be finished. At this point, it is not working (I have read your message though). Anyways, since the structure implemented looks all right and you are already working on the milestone 3 I will approve it.

Regards.

-----------------------------------

* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

**Pull request closed without merging**